rfcs/proposed/task_group_dynamic_dependencies/implementation_details.md

```diff
@@ -0,0 +1,754 @@
+* When `transfer_successors_to(new_task)` is called from the running task, a dynamic state is created for `new_task`.
+
+If the associated dynamic state was already initialized, `get_dynamic_state` returns it.
+
+If multiple threads concurrently perform the actions above on the same task instance (e.g. concurrently adding
+successors to the same predecessor before its dynamic state is created), each thread allocates a new dynamic state object
+and attempts to update the atomic object in `task_handle_task` using a `compare_exchange_strong` operation. If the operation fails, meaning
+another thread has already stored the dynamic state, the state allocated by the current thread is destroyed:
+
+The created `task_dynamic_state` object initially holds a reference counter to prolong its lifetime
+while the task is in progress. When the task is destroyed, this reference counter is decreased.
+
+## `tbb::task_tracker` class implementation
+
+The `tbb::task_tracker` class represents a task for the purpose of creating task dependencies. It can be constructed or assigned from a
+`task_handle`. It is copyable and movable, with copies referring to the same `task_dynamic_state` object.
+
+The dynamic state of a task remains valid until both the task and all associated `task_tracker` objects are destroyed.
+
+A default-constructed `tbb::task_tracker` does not represent any task.
+
+```cpp
+class task_tracker {
+private:
+    task_dynamic_state* m_state;
+public:
+    task_tracker() : m_state(nullptr) {}
+
+    task_tracker(const task_tracker& other) : m_state(other.m_state) {
+        if (m_state) m_state->reserve();
+    }
+    task_tracker& operator=(const task_tracker& other) {
+        if (this != &other) {
+            if (m_state) m_state->release();
+            m_state = other.m_state;
+            if (m_state) m_state->reserve();
+        }
+        return *this;
+    }
+
+    task_tracker(task_tracker&& other) : m_state(other.m_state) {
+        other.m_state = nullptr;
+    }
+    task_tracker& operator=(task_tracker&& other) {
+        if (this != &other) {
+            if (m_state) m_state->release();
+            m_state = other.m_state;
+            other.m_state = nullptr;
+        }
+        return *this;
+    }
+
+    task_tracker(const task_handle& th) : m_state(th ? th.get_dynamic_state() : nullptr) {
+        if (m_state) m_state->reserve();
+    }
+    task_tracker& operator=(const task_handle& th) {
+        if (m_state) m_state->release();
+
+        if (th) {
+            m_state = th.get_dynamic_state();
+            m_state->reserve();
+        } else {
+            m_state = nullptr;
+        }
+        return *this;
+    }
+
+    ~task_tracker() { if (m_state) m_state->release(); }
+};
+```
+
+## `task_dynamic_state` in details
+
+As mentioned above, the `task_dynamic_state` class tracks a task's state (completed or not),
+its list of successors, and any post-transfer actions. It has the following layout:
```

Comment:
What is meant by "post-transfer" actions here? Is it about using the `m_new_dynamic_state` pointer after `transfer_successors_to` was made?